tequilapi/endpoints/identities.go

```diff
@@ -91,6 +98,57 @@ func (endpoint *identitiesAPI) List(resp http.ResponseWriter, request *http.Requ
 	utils.WriteAsJSON(idsSerializable, resp)
 }
 
+// swagger:operation PUT /me Identity myIdentity
```

Comment:
It's not very RESTish. A better url would be /identities/current or smth like that. It's subresourse of identities

Reply:
Yeah forgot to mention :D https://github.com/julienschmidt/httprouter/issues/73 couldn't do it due to a limitation in our router package :/

Reply:
basically you can't have `identities/x` if you have `identities/:id`

Reply:
I don't see how its related

Reply:
Ok I see it. Time to migrate to [Gin](https://github.com/gin-gonic/gin)

Reply:
Yeah, that'd be a good choice:)

Reply:
It even can be handled inside identities controller with if identity == "current" { ... } it's ugly hack and needs //TODO at least, but it hides limitations of framework especially in these sensitive parts as external api definition. This way we can have nice API and do framework migration after without external changes.

Reply:
Moving to gin will not solve this. Gin is basicly using a custom version of julienschmidt/httprouter. They've just baked it in. Trade off is that you sacrifice speed of routing for the lack of customizability. Gin might not be the web server we need here.

This will panic:

```golang
package main

import (
	"net/http"

	"github.com/gin-gonic/gin"
)

func main() {
	router := gin.Default()
	router.GET("/identity", func(c *gin.Context) {
		c.String(http.StatusOK, "/identity")
	})
	router.GET("/identity/x", func(c *gin.Context) {
		c.String(http.StatusOK, "/identity/x/")
	})
	router.GET("/identity/:id", func(c *gin.Context) {
		c.String(http.StatusOK, "/identity/:id/")
	})
	router.Run(":1234")
}
```

output:
```
panic: wildcard route ':id' conflicts with existing children in path '/identity/:id'
```

Reply:
I have used this not long ago and it worked for this case:
`https://github.com/gorilla/mux`

Reply:
Changed the endpoint url to /identities/current